### PR TITLE
Add text content renderer

### PR DESCRIPTION
**Pull request by JinneeJ**
Finally added ability to get clean text.

P.S.: Really sorry for big delay 🙏

### CODE REVIEW

**Comment by robinst** [comment]
Nice change!

Do you have any plans for making it work for custom nodes for extensions (e.g. `Strikethrough`)? Maybe we should have a similar model as for HTML rendering where extensions can contribute code for rendering, and users can customize the rendering for specific node types. See `NodeRenderer` and friends.

(By the way, I was on holidays, that's why it took so long to respond.)

File: commonmark/src/main/java/org/commonmark/content/TextContentRenderer.java [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+                    textContent.write("- ");
+                }
+                visitChildren(listItem);
+                writeEndOfLine(listItem, null);
+            }
+        }
+
+        @Override
+        public void visit(OrderedList orderedList) {
+            orderedListCounter = 1;
```
Comment (JinneeJ) [inline commonmark/src/main/java/org/commonmark/content/TextContentRenderer.java]: You're right :) Fixed here https://github.com/atlassian/commonmark-java/pull/58/commits/2d520cc8736c0b0f078f0ffa529dabd479c4ef66

**Comment by JinneeJ** [comment]
Thanks for review! I've made the fix with lists.

Also I've tried to implement support extensions for TextContentRenderer. But I had to make some changes in core in order to avoid copying of code. You can see it in last 5 commits. Let me know what you think about those changes 😊

**Comment by robinst** [comment]
Cool, starting to look good!

I wanted to make a few suggestions but wasn't sure how they'd work, so I checked out the branch and started modifying it :). I merged master into your branch and did these changes on top:

1. https://github.com/atlassian/commonmark-java/commit/43c7e28a7cd32d5fb5a4fb367f4bf7dd7371d627
2. https://github.com/atlassian/commonmark-java/commit/91827e2364f9f3bb162df22cf012d04f6c0980fb

Can you have a look and tell me if it makes sense?

**Comment by JinneeJ** [comment]
I think your changes are great 😊

**Comment by robinst** [comment]
Thanks! Merged now. Will do a release soon with these and other changes.

**Comment by robinst** [comment]
Released now, thanks a lot @JinneeJ: https://github.com/atlassian/commonmark-java/releases/tag/commonmark-parent-0.7.0

**Pull request merged**